### ISSUE TITLE
Upgrade to debian:10 for latest version of qmk

### ISSUE THREAD

**Issue by kkrauth** [qmk/qmk_base_container#3]
Would suggest updating the base image to debian:10 as it will fetch python3.7 by default (instead of 3.5 by debian:9), required by new versions of QMK. 

Also added "systemd" as another apt-get dependency to be downloaded as it's used by qmk-doctor. I was able to run the latest qmk successfully in the container using the following dockerfile:

```
FROM debian:10

RUN apt-get update && apt-get install --no-install-recommends -y \
    avr-libc \
    avrdude \
    binutils-arm-none-eabi \
    binutils-avr \
    build-essential \
    dfu-programmer \
    dfu-util \
    gcc \
    gcc-avr \
    git \
    libnewlib-arm-none-eabi \
    python3 \
    python3-pip \
    python3-setuptools \
    software-properties-common \
    systemd \
    unzip \
    wget \
    zip \
    && rm -rf /var/lib/apt/lists/*

# upgrade gcc-arm-none-eabi from the default 5.4.1 to 6.3.1 due to ARM runtime issues
RUN wget -q https://developer.arm.com/-/media/Files/downloads/gnu-rm/6-2017q2/gcc-arm-none-eabi-6-2017-q2-update-linux.tar.bz2 -O - | \
    tar xj --strip-components=1 -C /
```

**Comment by zvecr** [comment]
Per https://github.com/qmk/qmk_firmware/issues/5776, QMK is targeting Python > 3.5 so the dependency upgrade is not mandatory at the moment. As for `systemd`, it would be nice to not include it as partially goes against the core concepts of containers.

The primary target of `qmk_base_container` is to unify the compilation environment for TravisCI and the Configurator. I'm not sure this extends at the moment to testing the QMK CLI.

**Comment by kkrauth** [comment]
I see, thanks for the explanation! Running /vagrant/bin/qmk inside the default docker container on the latest source resulted in:
```
TypeError: __init__() got an unexpected keyword argument 'capture_output'
```
Which was addressed with the debian update and python3.7. 

However, if the intent of this base image is not to support running the CLI inside the container, then I'll close it. 

In case someone else runs across this issue, also had to install the two missing modules:
```
python3 -m pip install argcomplete
python3 -m pip install colorama
```
That belongs in the qmk_firmware Dockerfile though.

**Comment by zvecr** [comment]
Its good info to have, as we have had a previous issue https://github.com/qmk/qmk_firmware/pull/6506 and we are always wanting to track down bugs and make the QMK cli bulletproof.

As qmk_firmware's Vagrant setup currently uses the Docker container, it could be a use case for another layer above the base_container for dev purposes.

Upgrading to version 10 should still be on the roadmap so an awesome thing is that, at least part of updating to Debian 10 has now been looked into. Thanks!

**Comment by skullydazed** [comment]
I just merged #6592 which should fix the python 3.5 related errors. #6588 will address the systemd dependency, making that check optional.
